uk.co.compucorp.civicrm.hrcore/CRM/HRCore/Menu/CiviAdapter.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+use CRM_HRCore_Menu_Item as MenuItem;
+
+class CRM_HRCore_Menu_CiviAdapter {
+
+  /**
+   * Builds the navigation menu tree and returns in a format expected
+   * by Civi in order to generate the navigation menu. It accepts a
+   * Menu Item instance
+   *
+   * @param MenuItem $menuItem
+   *
+   * @return array
+   *  Sample return format:
+   *  [
+   *    'attributes' =>
+   *      [
+   *        'label' => 'Components',
+   *         'url' => 'civicrm/bla',
+   *         'permission' => 'sample permission',
+   *         'operator' => 'AND',
+   *      ],
+   *   ],
+   */
+  public function getNavigationTree(MenuItem $menuItem) {
+    $navID = 1;
```

Comment:
Would there be a difference if it was 0 instead of 1? How about setting this as the default value of `buildMenuTree()`, the same way that the default value for parent id is 0?

Also, I've just realized that you're using two different names for the same thing. Here you call it "Navigation Tree" but on the other method you call it "Menu Tree". Lets use the same name for both

Reply:
If the navID was set to Zero, Top level menu items having nav ID as zero will have their direct children having the parentID as 0, same as their parent (top level item also have parentId as 0) which will be conflicting.
The reason why I did this way is to mimic Civi that sets the parent ID of top navigation items to NULL but each navigation Item has its own unique ID.